cds-snc/forms-terraform
# Datalake V1

Adds:

* AWS S3 Buckets: Provisioned for storing raw [Historical Templates from Air Table & Foundational for any other Excel or JSON Extracts from external systems] and processed data from the RDS.

* AWS Glue Jobs: Configured for data processing tasks, including extraction, transformation, and loading (ETL) operations. Uses Python and Apache Spark/AWS Specific Glue Services for Performant Transforms

To Test:

As this adds the GLUE and ATHENA modules, you must rebuild the container for deploy. This also updates Localstack to 4.1.
As this requires a build of the container, ensure the App is launched to build the DB.
Athena & Glue Jobs can be accessed via Localstack [ https://app.localstack.cloud/inst/default/resources/glue/jobs ] & [ https://app.localstack.cloud/inst/default/resources/athena/sql ]
On Localstack Cron jobs won't occur, so you must manually execute the jobs. 

To Execute Glue:
1. Go to the Localstack Web Instance, Run the historical or rds jobs.
2. Swap to Crawlers, run the equivalent for either of the runs of jobs.
3. Go to Athena, and try an SQL Query on the dataset!~

Notes:
Terragrunt Caches may cause issues, you can nuke them with:

```
find ./ -type d -name ".terragrunt-cache" -exec rm -rf {} +  
find ./ -type f -name ".terraform.lock.hcl" -exec rm -rf {} +
find ./ -type f -name ".terraform.tfstate" -exec rm -rf {} + 
terragrunt run-all init --upgrade   
```

If an error occurs with the ETL job, you can access the docker log to see the error message (surrounded by tons of java errors)
